src/main/kotlin/org/rust/lang/core/resolve/ref/RsStructExprFieldReferenceImpl.kt

```diff
@@ -18,6 +18,8 @@ class RsStructLiteralFieldReferenceImpl(
 ) : RsReferenceCached<RsStructLiteralField>(field),
     RsReference {
 
+    var renameSourceType: RenameSourceType? = null
```

Comment:
Mutable state in reference looks rather suspiciously

Reply:
If I can I would add it as constructor argument, but I cannot pass that value where `RsStructLiteralFieldReferenceImpl` object is created (`org.rust.lang.core.psi.ext.RsStructLiteralFieldImplMixin#getReference`), that's why I considered to add that mutable property

Reply:
Can you suggest how can I pass info about rename source (field or binding) to handleRename without carrying it in mutable property?

Reply:
I would suggest moving all rename logic from `RsStructLiteralFieldReferenceImpl.handleElementRename()` to `RsRenameProcessor.renameElement()`, where it is possible to get access to rename source and then rename `RsStructLiteralField`s accordingly.